llvm/include/llvm/Support/Compiler.h

```diff
@@ -189,19 +190,23 @@
 #define LLVM_TEMPLATE_ABI __declspec(dllimport)
 #define LLVM_EXPORT_TEMPLATE
 #endif
+#define LLVM_ABI_EXPORT __declspec(dllexport)
```

Comment:
Why is this being added? I'm weary of any macro that is always export only because the value needs to switch between the definition and reference.

Reply:
Its macro I mentioned earlier initial explanation about being used in Clang and LLVM, it should never be used for things that could be references.

Reply:
I think this is subtle enough that we may need some documentation comments explaining when and how someone should use this. (Most reviewers aren't going to be well-versed in how shared libraries work on Windows.)

Reply:
I added a comment to try and explains its use and behaviour.